airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -948,7 +948,10 @@ def execute(self, context) -> None:
                 delegate_to=self.delegate_to,
                 impersonation_chain=self.impersonation_chain,
             )
-            schema_fields = json.loads(gcs_hook.download(gcs_bucket, gcs_object).decode("utf-8"))
+            schema_fields_bytes_or_string = gcs_hook.download(gcs_bucket, gcs_object)
+            if hasattr(schema_fields_bytes_or_string, 'decode'):
+                schema_fields_bytes_or_string = cast(bytes, schema_fields_bytes_or_string).decode("utf-8")
+            schema_fields = json.loads(schema_fields_bytes_or_string)
         else:
```

Comment:
I recently changed `gcs_hooks`'s return value as the solution to an issue.
So if you are not passing the `filename` to the `download()` method it will return only bytes otherwise file name.
Therefore the variable `schema_fields_bytes_or_string` always will be as bytes, but unfortunately `gcs_hook.download`'s return type is `bytes | str`.

So how do you think could we skip this check `if hasattr(schema_fields_bytes_or_string, 'decode')`?

Also, maybe we could use here something like that?
`gcs_hook.get_conn().bucket(gcs_bucket).blob(blob_name=gcs_object).download_as_bytes()`

Reply:
Good point. I changed it in the way that I created a separate method in the hook that only returns bytes (and does not pass filename) and cast it to `bytes` - with appropriate comment.